editor/editor_node.cpp

```diff
@@ -3727,10 +3728,15 @@ bool EditorNode::is_scene_in_use(const String &p_path) {
 	return false;
 }
 
+void EditorNode::register_editor_paths(bool p_for_project_manager) {
+	memnew(EditorPaths(p_for_project_manager));
```

Comment:
I'm no fan of this pattern. At the very least we should add a `CRASH_COND(singleton != nullptr)` to the constructor in order to avoid leaks. But even then I find this confusing to read.

Reply:
okay, let me fix it.